src/containers/LoginForm.js

```diff
@@ -18,12 +19,17 @@ class LoginForm extends Component {
     this.setState({ [e.target.name]: e.target.value });
   };
 
-  handleSubmit = async () => {
-    // e.preventDefault();
+  handleSubmit = async e => {
+    e.preventDefault();
     const { email, password } = this.state;
-    let user = await fetchUser(email, password);
-    if (user) {
+    try {
+      let user = await fetchUser(email, password);
       this.props.signIn(user);
+      this.props.history.push('/');
+      this.props.hasErrored('');
+    } catch ({ message }) {
+      this.props.hasErrored(message);
+      this.props.history.push('/login');
```

Comment:
Everything is looking great @vjt960!  I'm curious about this.props.hisotry.push('./login), does this store the history of the pages they've viewed for redirection purposes?

Reply:
**I tried making it work with the NavLink, but...**

![](https://qph.fs.quoracdn.net/main-qimg-13c446615040778bc8de53e343986ec4)

Reply:
Gotcha, well it seems to work well, I just haven't seen that before!  So...

![](https://media.giphy.com/media/pCO5tKdP22RC8/giphy.gif)

Reply:
Very interesting! Would you explain that to me more in depth the next time we're in person?